core/src/cluster_slots.rs

```diff
@@ -141,14 +140,21 @@ impl ClusterSlots {
             self.epoch = Some(epoch);
         }
     }
-    pub fn peers(&self, slot: Slot) -> Vec<(Rc<Pubkey>, u64)> {
-        let mut peers: HashMap<Rc<Pubkey>, u64> = self.validator_stakes.clone();
-        if let Some(slot_peers) = self.lookup(slot) {
-            slot_peers
-                .iter()
-                .for_each(|(x, y)| *peers.entry(x.clone()).or_insert(0) += *y);
-        }
-        peers.into_iter().filter(|x| *x.0 != self.self_id).collect()
+
+    pub fn best_peer(&self, slot: Slot, repair_peers: &[ContactInfo]) -> usize {
+        let slot_peers = self.lookup(slot);
+        let weights: Vec<(u64, usize)> = repair_peers
+            .iter()
+            .enumerate()
+            .map(|(i, x)| {
+                (
+                    1 + slot_peers.and_then(|v| v.get(&x.id)).cloned().unwrap_or(0)
```

Comment:
So basically if you've voted for it, you're counted 2x your stake, if not, you're only counted 1x?

Reply:
So if 33% of the stake has advertised they have the slot, then they're counted as much as the rest of the 66%, so 1/2 of the requests should go to the people with the slot?

Reply:
Yea. Otherwise I think fake votes could hurt repair. The default behavior was picking at a linear random.